vspace/src/main/java/edu/asu/diging/vspace/core/factory/impl/SpaceFactory.java

```diff
@@ -1,23 +1,60 @@
 package edu.asu.diging.vspace.core.factory.impl;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.thymeleaf.util.StringUtils;
 
+import edu.asu.diging.vspace.core.data.ExhibitionLanguageRepository;
+import edu.asu.diging.vspace.core.data.LocalizedTextRepository;
+import edu.asu.diging.vspace.core.data.SpaceRepository;
 import edu.asu.diging.vspace.core.factory.ISpaceFactory;
 import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.model.impl.ExhibitionLanguage;
+import edu.asu.diging.vspace.core.model.impl.LocalizedText;
 import edu.asu.diging.vspace.core.model.impl.Space;
+import edu.asu.diging.vspace.web.staff.forms.LocalizedTextForm;
 import edu.asu.diging.vspace.web.staff.forms.SpaceForm;
 
 @Service
 public class SpaceFactory implements ISpaceFactory {
+    
+    @Autowired
+    private SpaceRepository spaceRepo;
+    
+    @Autowired
+    private ExhibitionLanguageRepository exhibitionLanguageRepository;
+    
+    @Autowired
+    private LocalizedTextRepository localizedTextRepo;
+    
 
-	/* (non-Javadoc)
-	 * @see edu.asu.diging.vspace.core.factory.impl.ISpaceFactory#createSpace(edu.asu.diging.vspace.web.staff.forms.SpaceForm)
-	 */
-	@Override
-	public ISpace createSpace(SpaceForm form) {
-		ISpace space = new Space();
-		space.setName(form.getName());
-		space.setDescription(form.getDescription());
-		return space;
-	}
+    @Override
+    public ISpace createSpace(SpaceForm form) {
+        ISpace space = new Space();
+        space.setName(form.getName());
+        space.setDescription(form.getDescription());
+        return spaceRepo.save((Space) space);
```

Comment:
it definitely needs to be documented (javadoc) that this method now stores the new space as well. that's a pretty big change.

Reply:
Actually, the factory shouldn't store the space. That's the job of the manager.